radix-engine/src/system/system.rs

```diff
@@ -110,6 +120,93 @@ impl TryFrom<ObjectHandle> for ActorObjectType {
     }
 }
 
+pub struct SchemaCache {
+    cache: NonIterMap<Hash, ScryptoSchema>,
+}
+
+impl SchemaCache {
+    pub fn new() -> Self {
+        Self {
+            cache: NonIterMap::new(),
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+pub enum KeyValueStoreSchemaIdent {
+    Key,
+    Value,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+pub enum FunctionSchemaIdent {
+    Input,
+    Output,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, ScryptoSbor)]
+pub enum BlueprintSchemaIdent {
+    Field(u8),
+    KeyValueStore {
+        collection_index: u8,
+        schema_ident: KeyValueStoreSchemaIdent,
+    },
+    Function {
+        ident: String,
+        schema_ident: FunctionSchemaIdent,
+    },
+    Event(String),
+}
```

Comment:
Alternatively, we can just pass `TypePointer` to `validate_payload_against_blueprint_schema`.

And type pointers can be obtained from method of `BlueprintInterface`, such as `BlueprintInterface::get_function_input_type(&self) -> TypePointer`.

Reply:
great idea!